toolchain/check/pattern.cpp

```diff
@@ -94,4 +96,41 @@ auto AddBindingPattern(Context& context, SemIR::LocId name_loc,
   return {.pattern_id = binding_pattern_id, .bind_id = bind_id};
 }
 
+// Returns a VarStorage inst for the given `var` pattern. If the pattern
+// is the body of a returned var, this reuses the return slot, and otherwise it
+// adds a new inst.
+static auto GetOrAddVarStorage(Context& context, SemIR::InstId var_pattern_id,
+                               bool returned) -> SemIR::InstId {
+  if (returned) {
+    auto& function = GetCurrentFunctionForReturn(context);
+    auto return_info =
+        SemIR::ReturnTypeInfo::ForFunction(context.sem_ir(), function);
+    if (return_info.has_return_slot()) {
+      return GetCurrentReturnSlot(context);
+    }
+  }
+  auto pattern = context.insts().GetWithLocId(var_pattern_id);
+
+  return AddInstWithCleanup(
+      context, pattern.loc_id,
+      SemIR::VarStorage{.type_id = ExtractScrutineeType(context.sem_ir(),
+                                                        pattern.inst.type_id()),
+                        .pattern_id = var_pattern_id});
+}
+
+auto AddPatternVarStorage(Context& context, SemIR::InstBlockId pattern_block_id,
+                          bool returned) -> void {
+  // We need to emit the VarStorage insts early, because they may be output
+  // arguments for the initializer. However, we can't emit them when we emit
+  // the corresponding `VarPattern`s because they're part of the pattern match,
+  // not part of the pattern.
+  // TODO: find a way to do this without walking the whole pattern block.
```

Comment:
```suggestion
  // TODO: Find a way to do this without walking the whole pattern block.
```

(I believe we typically capitalize in this TODO case)

Reply:
Done. (This was just moved from a different file.)